src/content/api/parser.md

```diff
@@ -37,279 +38,477 @@ as such:
 
 `SyncBailHook`
 
-Evaluate the type of an identifier.
+Triggered when evaluating an expression consisting in a `typeof` of a free variable
 
-Parameters: `expression`
+- Hook Parameters: `identifier`
+
+- Callback Parameters: `expression`
+
+```js
+parser.hooks.evaluateTypeof.for('myIdentifier').tap('MyPlugin', expression => {
+  /* ... */
+  return expressionResult;
+});
+```
+
+```js
+// This will trigger the evaluateTypeof hook
+const a = typeof myIdentifier;
+```
+
+```js
+// This won't
+const myIdentifier = 0;
+const b = typeof myIdentifier;
+```
 
 
 ### evaluate
 
 `SyncBailHook`
 
-Evaluate an expression.
+Called when evaluating an expression.
 
-Parameters: `expression`
+- Hook parameters: `expressionType`
+
+- Callback parameters: `expression`
+
+```js
+parser.hooks.evaluate.for(/* expression type */).tap(/* ... */);
+```
+
+Where the expressions types are:
+
+- `'ArrowFunctionExpression'`
+- `'AssignmentExpression'`
+- `'AwaitExpression'`
+- `'BinaryExpression'`
+- `'CallExpression'`
+- `'ClassExpression'`
+- `'ConditionalExpression'`
+- `'FunctionExpression'`
+- `'Identifier'`
+- `'LogicalExpression'`
+- `'MemberExpression'`
+- `'NewExpression'`
+- `'ObjectExpression'`
+- `'SequenceExpression'`
+- `'SpreadElement'`
+- `'TaggedTemplateExpression'`
+- `'TemplateLiteral'`
+- `'ThisExpression'`
+- `'UnaryExpression'`
+- `'UpdateExpression'`
 
 
 ### evaluateIdentifier
 
 `SyncBailHook`
 
-Evaluate an identifier that is a free variable.
+Called when evaluating an identifier that is a free variable.
 
-Parameters: `expression`
+- Hook Parameters: `identifier`
+
+- Callback Parameters: `expression`
 
 
 ### evaluateDefinedIdentifier
 
 `SyncBailHook`
 
-Evaluate an identifier that is a defined variable.
+Called when evaluating an identifier that is a defined variable.
+
+- Hook Parameters: `identifier`
 
-Parameters: `expression`
+- Callback Parameters: `expression`
 
 
 ### evaluateCallExpressionMember
 
 `SyncBailHook`
 
-Evaluate a call to a member function of a successfully evaluated expression.
+Called when evaluating a call to a member function of a successfully evaluated expression.
 
-Parameters: `expression` `param`
+- Hook Parameters: `identifier`
+
+- Callback Parameters: `expression` `param`
+
+```js
+// This will trigger the hook for('myFunc')
```

Comment:
Can we move the comment out of the js block since it is also single purpose example?